tests/unittests/sources/test_hetzner.py

```diff
@@ -114,29 +113,28 @@ def test_read_data(
             ],
         )
 
-        self.assertTrue(m_readmd.called)
+        assert m_readmd.called
 
-        self.assertEqual(METADATA.get("hostname"), ds.get_hostname().hostname)
+        assert METADATA.get("hostname") == ds.get_hostname().hostname
 
-        self.assertEqual(METADATA.get("public-keys"), ds.get_public_ssh_keys())
+        assert METADATA.get("public-keys") == ds.get_public_ssh_keys()
 
-        self.assertIsInstance(ds.get_public_ssh_keys(), list)
-        self.assertEqual(ds.get_userdata_raw(), USERDATA)
-        self.assertEqual(ds.get_vendordata_raw(), METADATA.get("vendor_data"))
+        assert isinstance(ds.get_public_ssh_keys(), list)
+        assert ds.get_userdata_raw() == USERDATA
+        assert ds.get_vendordata_raw() == METADATA.get("vendor_data")
 
     @mock.patch("cloudinit.sources.helpers.hetzner.read_metadata")
     @mock.patch("cloudinit.net.find_fallback_nic")
     @mock.patch("cloudinit.sources.DataSourceHetzner.get_hcloud_data")
     def test_not_on_hetzner_returns_false(
-        self, m_get_hcloud_data, m_find_fallback, m_read_md
+        self, m_get_hcloud_data, m_find_fallback, m_read_md, ds
     ):
         """If helper 'get_hcloud_data' returns False,
         return False from get_data."""
         m_get_hcloud_data.return_value = (False, None)
-        ds = self.get_ds()
         ret = ds.get_data()
 
-        self.assertFalse(ret)
+        assert not ret
         # These are a white box attempt to ensure it did not search.
         m_find_fallback.assert_not_called()
         m_read_md.assert_not_called()
```

Comment:
Aligned with rtd/development/testing.rst can we replace assert_not_called() with ` assert not called` (and maybe update doc/rst/development/testing.rst) replacing 
  * ``m.assert_not_called()`` => ``assert 0 == m.call_count``
  
with 
  * ``m.assert_not_called()`` => ``assert not called`